### PR TITLE
Update the gallery example for plotting lines with LineString/MultiLineString geometry

### LINKED ISSUE #3320: Use a different dataset in the gallery example "examples/gallery/lines/roads.py"
The gallery example (https://www.pygmt.org/v0.12.0/gallery/lines/roads.html) uses a dataset from http://www2.census.gov/geo/tiger/TIGER2015/PRISECROADS/tl_2015_15_prisecroads.zip.

I'm getting the "Access Denied" error when I try to download the data using a China IP, but it works when I use a VPN server in US or Japan.

It would be better if we could find another dataset which is more accessible to users.

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

Fixes  #3320 

**Preview**: https://pygmt-dev--3711.org.readthedocs.build/en/3711/gallery/lines/linestrings.html#

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash command is:

- `/format`: automatically format and lint the code

### CODE REVIEW

**Comment by seisman** [comment]
Two thoughts:

1. Perhaps we should have [`geodatasets`](https://github.com/geopandas/geodatasets) installed in our CI, so that we don't have to have the long URL hardcoded.
2. The title "Rivers" gives me the expression that this is about a river dataset, but actually it's all about plotting LineString/MultiLineString geometries in GeoPandas objects (or geojson format). Maybe rename the title to something like "Plotting lines with LineString/MultiLineString geometry", "GeoPandas: Lines with LineString/MultiLineString geometry"

**Comment by michaelgrund** [comment]
> Two thoughts:
> 
>     1. Perhaps we should have [`geodatasets`](https://github.com/geopandas/geodatasets) installed in our CI, so that we don't have to have the long URL hardcoded.
> 
>     2. The title "Rivers" gives me the expression that this is about a river dataset, but actually it's all about plotting LineString/MultiLineString geometries in GeoPandas objects (or geojson format). Maybe rename the title to something like "Plotting lines with LineString/MultiLineString geometry", "GeoPandas: Lines with LineString/MultiLineString geometry"

Agree with both points. Regarding point 1 we just want to add it for the docs/gallery example and not as an general dependency, right?

**Comment by seisman** [comment]
> > Two thoughts:
> > ```
> > 1. Perhaps we should have [`geodatasets`](https://github.com/geopandas/geodatasets) installed in our CI, so that we don't have to have the long URL hardcoded.
> > 
> > 2. The title "Rivers" gives me the expression that this is about a river dataset, but actually it's all about plotting LineString/MultiLineString geometries in GeoPandas objects (or geojson format). Maybe rename the title to something like "Plotting lines with LineString/MultiLineString geometry", "GeoPandas: Lines with LineString/MultiLineString geometry"
> > ```
> 
> Agree with both points. Regarding point 1 we just want to add it for the docs/gallery example and not as an general dependency, right?

Yes.

**Comment by yvonnefroehlich** [comment]
Regarding point 1, this would be also relevant / helpful for the Gallery example [Choropleth map](https://www.pygmt.org/dev/gallery/maps/choropleth_map.html).

**Comment by seisman** [comment]
> Regarding point 1, this would be also relevant / helpful for the Gallery example [Choropleth map](https://www.pygmt.org/dev/gallery/maps/choropleth_map.html).

Maybe we should add `geodatasets` and update the Choropleth example in a separate PR first.

**Comment by seisman** [comment]
> > Regarding point 1, this would be also relevant / helpful for the Gallery example [Choropleth map](https://www.pygmt.org/dev/gallery/maps/choropleth_map.html).
> 
> Maybe we should add `geodatasets` and update the Choropleth example in a separate PR first.

I'll open a PR in minites.

File: examples/gallery/lines/linestrings.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# ruff: noqa: RUF003
```
Comment (seisman) [inline examples/gallery/lines/linestrings.py:1]:
```suggestion
```

File: examples/gallery/lines/linestrings.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+"""
+Plotting lines with LineString/MultiLineString geometry
+=====
```
Comment (seisman) [inline examples/gallery/lines/linestrings.py:4]:
```suggestion
Plotting lines with LineString/MultiLineString geometry
=======================================================
```

File: examples/gallery/lines/linestrings.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+# Add rivers to map
+fig.plot(data=gpd_lines_new, pen="1p,steelblue")
+
+fig.show()
+
+
+
+
+
+
```
Comment (yvonnefroehlich) [inline examples/gallery/lines/linestrings.py:50]: Please remove all these blank lines at the end of the file.

File: examples/gallery/lines/linestrings.py (hunk 1 above)
Comment (yvonnefroehlich) [reply examples/gallery/lines/linestrings.py:1]: @michaelgrund seems you applied @seisman's suggestion in commit https://github.com/GenericMappingTools/pygmt/pull/3711/commits/47dabbc540fa009172ac92da4469b5c3b6e4d8ca, but reverted it (accidently ?) in commit https://github.com/GenericMappingTools/pygmt/pull/3711/commits/1dbebb57d5c1345369cef7fb11668edea0aa0193 🙈.
```suggestion

```

File: examples/gallery/lines/linestrings.py (hunk 2 above)
Comment (yvonnefroehlich) [reply examples/gallery/lines/linestrings.py:4]: Same as above, but for commit https://github.com/GenericMappingTools/pygmt/pull/3711/commits/d222759b4041a982e730e88ba2da2ec1b057916f.

File: examples/gallery/lines/linestrings.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+# %%
+import geodatasets
+import geopandas as gpd
+import pygmt
+
+# Read shapefile data of large rivers in Europe using geopandas
+gdf = gpd.read_file(geodatasets.get_path("eea large_rivers"))
+
+# Convert object to EPSG 4326 coordinate system
+gpd = gpd.to_crs("EPSG:4326")
```
Comment (yvonnefroehlich) [inline examples/gallery/lines/linestrings.py:22]:
```suggestion
gdf = gdf.to_crs("EPSG:4326")
```

File: examples/gallery/lines/linestrings.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    region=[-10, 30, 35, 57],
+    resolution="l",
+    land="gray95",
+    shorelines="1/0.1p,gray50",
+    borders="1/0.1,gray30",
+    frame=True,
+)
+
+# Add rivers to map
+fig.plot(data=gpd, pen="1p,steelblue")
```
Comment (yvonnefroehlich) [inline examples/gallery/lines/linestrings.py:37]: For me, a thinner line looks better.
```suggestion
fig.plot(data=gdf, pen="0.5p,steelblue")
```

Comment (michaelgrund) [reply examples/gallery/lines/linestrings.py:37]: I would prefer 1p since it's the element on which the focus lies in this example.

Comment (yvonnefroehlich) [reply examples/gallery/lines/linestrings.py]: Hm. OK. Edited the suggestion to only include the fix from "gpd" to "gdf".

File: examples/gallery/lines/linestrings.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
+"""
+Plotting lines with LineString/MultiLineString geometry
```
Comment (seisman) [inline examples/gallery/lines/linestrings.py:2]: I'm wondering if it makes more sense to change the title to:
```
GeoPandas: Plotting lines with LineString/MultiLineString geometry
```

File: examples/gallery/lines/linestrings.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+=======================================================
+
+The :meth:`pygmt.Figure.plot` method allows us to plot geographical data such
+as lines which are stored in a :class:`geopandas.GeoDataFrame` object. Use
+:func:`geopandas.read_file` to load data from any supported OGR format such as
```
Comment (seisman) [inline examples/gallery/lines/linestrings.py:7]:
```suggestion
:func:`geopandas.read_file` to load data from any supported OGR formats such as
```

Comment (seisman) [inline examples/gallery/lines/linestrings.py:7]: Please wrap these lines to 88 characters.

File: examples/gallery/lines/linestrings.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+# %%
+import geodatasets
+import geopandas as gpd
+import pygmt
+
+# Read shapefile data of large rivers in Europe using geopandas
+gdf = gpd.read_file(geodatasets.get_path("eea large_rivers"))
+
+# Convert object to EPSG 4326 coordinate system
+gdf = gdf.to_crs("EPSG:4326")
```
Comment (seisman) [inline examples/gallery/lines/linestrings.py:22]: Check the content of the data before plotting.
```suggestion
gdf = gdf.to_crs("EPSG:4326")
print(gdf)
```

File: examples/gallery/lines/linestrings.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
+"""
+Plotting lines with LineString/MultiLineString geometry
+=======================================================
```
Comment (yvonnefroehlich) [inline examples/gallery/lines/linestrings.py:3]: We should add white spaces here:
```suggestion
Plotting lines with LineString / MultiLineString geometry
=========================================================
```

Without white spaces the title does not fit nicely on the card in the gallery:
gpd_lines_NO_whitespaces

File: examples/gallery/lines/linestrings.py (hunk 7 above)
Comment (michaelgrund) [reply examples/gallery/lines/linestrings.py:7]: I think format is correct here.

File: examples/gallery/lines/linestrings.py [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+# Read shapefile data of large rivers in Europe using geopandas
+gdf = gpd.read_file(geodatasets.get_path("eea large_rivers"))
+
+# Convert object to EPSG 4326 coordinate system
+gdf = gdf.to_crs("EPSG:4326")
+print(gdf)
+
+fig = pygmt.Figure()
+
```
Comment (yvonnefroehlich) [inline examples/gallery/lines/linestrings.py:27]: I would split the code into two parts via `# %%`, to have the table showing the GeoDataFrame displayed within the code and before the image. Using `.head()` displays only the first lines of the GeoDataFrame, which should be enough to get an idea about the data. Both `gdf.head()` and `print(gdf.head())` work in the Gallery (with a slightly different appearance). But for using `gdf.head()`, in "normal" Python scripts it's only displayed when running the subsection, but not when running the complete script. So, using `print(gdf.head())`is maybe better here.

```suggestion
# Convert object to EPSG 4326 coordinate system
gdf = gdf.to_crs("EPSG:4326")
print(gdf.head())

# %%
fig = pygmt.Figure()

```

File: examples/gallery/lines/linestrings.py [hunk 11]
```diff
@@ -0,0 +1,41 @@
+"""
+GeoPandas: Plotting lines with LineString/MultiLineString geometry
+==================================================================
```
Comment (yvonnefroehlich) [inline examples/gallery/lines/linestrings.py:3]:
```suggestion
GeoPandas: Plotting lines with LineString / MultiLineString geometry
====================================================================
```

Comment (seisman) [inline examples/gallery/lines/linestrings.py:3]:
```suggestion
GeoPandas: Plotting lines with LineString or MultiLineString geometry
=====================================================================
```

File: examples/gallery/lines/linestrings.py [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+The :meth:`pygmt.Figure.plot` method allows us to plot geographical data
+such as lines which are stored in a :class:`geopandas.GeoDataFrame` object.
+Use :func:`geopandas.read_file` to load data from any supported OGR format
+such as a shapefile (.shp), GeoJSON (.geojson), geopackage (.gpkg), etc.
+Then, pass the :class:`geopandas.GeoDataFrame` as an argument to the
+``data`` parameter of :meth:`pygmt.Figure.plot`, and style the geometry
+using the ``pen`` parameter.
```
Comment (seisman) [inline examples/gallery/lines/linestrings.py:11]:
```suggestion
The :meth:`pygmt.Figure.plot` method allows us to plot geographical data such as lines
with LineString or MultiLineString geometry types stored in a
:class:`geopandas.GeoDataFrame` object or any object that implements the
`__geo_interface__ <https://gist.github.com/sgillies/2217756>`__ property.

Use :func:`geopandas.read_file` to load data from any supported OGR format such as a
shapefile (.shp), GeoJSON (.geojson), geopackage (.gpkg), etc. Then, pass the
:class:`geopandas.GeoDataFrame` object as an argument to the ``data`` parameter of
:meth:`pygmt.Figure.plot`, and style the lines using the ``pen`` parameter.
```

File: examples/gallery/lines/linestrings.py [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+"""
+
+# %%
+import geodatasets
+import geopandas as gpd
+import pygmt
+
+# Read shapefile data of large rivers in Europe using geopandas
```
Comment (seisman) [inline examples/gallery/lines/linestrings.py:19]:
```suggestion
# Read a sample dataset provided by the geodatasets package.
# The dataset contains large rivers in Europe, stored as LineString/MultiLineString
# geometry types.
```

File: examples/gallery/lines/linestrings.py (hunk 11 above)
Comment (yvonnefroehlich) [reply examples/gallery/lines/linestrings.py:3]: See suggestion by @seisman using "or" instead of a "/".

**Comment by seisman** [comment]
I've committed the suggestion @yvonnefroehlich and I made, so that this PR can be merged in v0.14.0.

**Pull request merged**